### PR TITLE
BDOG-632:  improve logging of file-reference

### PR DESCRIPTION
**Pull request by nigelhp**

### CODE REVIEW

File: app/connectors/aws/S3EventParser.scala [hunk 1]
```diff
@@ -67,21 +67,9 @@ class S3EventParser @Inject()(implicit ec: ExecutionContext) extends MessagePars
...
-        // manually touch the MDC, since subsequent log statements will have access to the file-reference and can thus
-        // use MdcLoggingExecutionContext / LoggingDetails.
-        try {
-          MDC.put("file-reference", event.location.objectKey)
-          logger.debug(s"Created FileUploadEvent for Key=[${event.location.objectKey}].")
-          Future.successful(event)
-        } finally {
-          MDC.remove("file-reference")
-        }
+        logger.debug(s"Created FileUploadEvent for bucket=[${event.location.bucket}] object=[${event.location.objectKey}].")
```
Comment (chotai) [inline app/connectors/aws/S3EventParser.scala:71]: Can you access `withLoggingDetails` here?

File: app/connectors/aws/S3EventParser.scala [hunk 2]
```diff
@@ -77,7 +77,7 @@ class S3EventParser @Inject()(implicit ec: ExecutionContext) extends MessagePars
         // use MdcLoggingExecutionContext / LoggingDetails.
         try {
           MDC.put("file-reference", event.location.objectKey)
```
Comment (chotai) [inline app/connectors/aws/S3EventParser.scala]: Do you have access to LoggingDetails?

**Review by chotai — APPROVED** [review]
Few places where `LoggingDetails` isn't used might be worth updating if possible

File: app/connectors/aws/S3EventParser.scala (hunk 2 above)
Comment (nigelhp) [reply app/connectors/aws/S3EventParser.scala]: This refers to an outdated version.  This direct use of MDC was removed in a later commit, as it was adding the wrong value as the 'fileReference'.  A new objectKey is allocated when verify moves the file post scan, and so objectKey != the original upload fileReference.

File: app/connectors/aws/S3EventParser.scala (hunk 1 above)
Comment (nigelhp) [reply app/connectors/aws/S3EventParser.scala:71]: Following a recent change, we could now use `withLoggingDetails`  here, but the MDC could only contain the `objectKey` (which is already logged in the message itself).  It cannot contain the `file-reference` as this is not contained in the event message - it has to be retrieved from metadata - which can only happen after this event parsing has completed.  The `objectKey` is not typically added to the MDC elsewhere - support was only added to retain some context for any exception thrown prior to the metadata being retrieved.

**Pull request merged**